<?xml version="1.0"?>
<launch>
    <!-- Launch file arguments -->
    <arg name="map"             default="floor2"/>
    <arg name="is_sim"          default="true"/>
    <arg name="use_foxglove"    default="true"/>
    <arg name="obstacle_map"    default="obstacles"/>
    <arg name="initial_pose_x"  default="-2.65488696"/>
    <arg name="initial_pose_y"  default="-1.64422277"/>
    <arg name="initial_pose_a"  default="1.57" /> <!-- wrt to map-->

    <!-- Start map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find svea_core)/maps/$(arg map).yaml" output="screen"/>

    <rosparam command="load" file="$(find svea_core)/params/$(arg obstacle_map).yaml" />

    <!-- If is_sim equal to false, then include all these tags-->
    <group unless="$(arg is_sim)">
        <!-- Start low-level interface -->
        <node name="serial_node" pkg="rosserial_python" type="serial_node.py">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="250000"/>
        </node>

        <!-- Start localization -->
        <include file="$(find svea_sensors)/launch/localize.launch">
            <arg name="initial_pose_x" value="$(arg initial_pose_x)" />
            <arg name="initial_pose_y" value="$(arg initial_pose_y)" />
            <arg name="initial_pose_a" value="$(arg initial_pose_a)" />
        </include>
    </group>

    <!-- Start Foxglove -->
    <include if="$(eval use_foxglove)" file="$(find foxglove_bridge)/launch/foxglove_bridge.launch">
        <arg name="port" value="8765" />
    </include>

    <!-- Start pure_pursuit -->
    <node name="pure_pursuit" pkg="svea_examples" type="pure_pursuit.py" output="screen">
        <param name="use_rviz" value="$(arg use_foxglove)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <!-- Waypoints for floo2 map -->
        <rosparam>
            state: [-7.4, -15.3, 0.9] # initial state (x, y, yaw)
            points:
            - [-2.3, -7.1]  # Bottom right (project room)
            - [10.5, 11.7]  # Top right
            - [5.7, 15.0]   # Top left
            - [-7.0, -4.0]  # Bottom left
        </rosparam>
    </node>
</launch>
